nisystemlink/clients/core/_uplink/_json_model.py

```diff
@@ -0,0 +1,16 @@
+from pydantic import BaseModel, Extra
+
+
+def _camelcase(s: str) -> str:
+    """Convert a snake case string to camelCase."""
+    parts = iter(s.split("_"))
+    return next(parts) + "".join(i.title() for i in parts)
+
+
+class JsonModel(BaseModel):
+    """Base class for models that are serialized to and from JSON."""
+
+    class Config:
+        alias_generator = _camelcase
+        allow_population_by_field_name = True
+        extra = Extra.forbid
```

Comment:
The `Extra.forbid` config makes it so Pydantic will error if the model receives an unknown field when initializing. I think this is valuable, because it will make our integration tests fail when we add new fields to any resources. It also will work nicely with the `ApiInfo` model below, causing errors if we add new operations without updating the client.

Reply:
I wonder if we can monkey patch the config to turn it on during integration tests. That would be the best of both worlds.

Reply:
That could be interesting, especially if it were an option to toggle it on or off. I could see cases where you make a change to the service that adds a new field and then someone wants to make an unrelated change to the client code and tests start failing.